### ISSUE TITLE
DynamicModel validate fail

### ISSUE THREAD

**Issue by berthojoris** [yiisoft/yii2#15153]
### What steps will reproduce the problem?
I create a simple code like 
```
$post = Yii::$app->request->post();

$model = DynamicModel::validateData($post, [
    [['mid', 'email'], 'required'],
    ['mid', 'number'],
    ['email', 'email']
]);

if ($model->hasErrors()) {
    return ["message" => "Validation fail. Please check your input!"];
}

return ["message" => "Validation pass!"];
```

### What is the expected result?
When I try to access my API by sending only **one** parameter (for example **just** a email parameter) I should get an `Validation fail. Please check your input!` error message from `$model->hasErrors()`.

### What do you get instead?
I get an error message `Getting unknown property: yii\\base\\DynamicModel::mid`

In this case does the `required` validation **just** check the parameter values? 

Does not check whether the parameters are sent or not? Do I have to make my own validation like `isset($_POST['mid'])`  before `DynamicModel::validateData`?

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.14-dev
| PHP version      | 7.1.11
| Operating system | Win

**Comment by papalapa** [comment]
Before validate you must assign $email and $mid with the same params in $_POST like this:
`$email = ArrayHelper::getValue($post, 'email');`
and
`$mid = ArrayHelper::getValue($post, 'mid');`

**Comment by michaelarnauts** [comment]
@maoxuner @berthojoris after more then 5 years, this has been fixed in the latest 2.0.48 release :)

**Comment by samdark** [comment]
Everything's possible in OpenSource when you take initiative :)
